pandas/tests/frame/test_subclass.py

```diff
@@ -210,3 +210,17 @@ def test_subclass_align_combinations(self):
         tm.assert_series_equal(res1, exp2)
         tm.assertIsInstance(res2, tm.SubclassedDataFrame)
         tm.assert_frame_equal(res2, exp1)
+
+    def test_subclass_sparse_slice(self):
+        ssdf = tm.SubclassedSparseDataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+        ssdf.testattr = "testattr"
+
+        tm.assertIsInstance(ssdf.loc[:2], tm.SubclassedSparseDataFrame)
```

Comment:
add a ``check_dtype`` option in testing function which will assert that not only they are instances of ``pd.SparseDataFrame`` , but also that they are the *same* (e.g. signature could be
``check_dtype=True|'same'``